compiler/tests/compile_errors.rs

```diff
@@ -8,6 +8,9 @@ use std::fs;
 #[rstest(
     fixture_file,
     error,
+    case("mismatch_return_type.fe", "[Str(\"semantic error: TypeError\")]"),
```

Comment:
@g-r-a-n-t 

I guess to satisfy this :point_up: test, I would have to add type checking here:

https://github.com/ethereum/fe/blob/de661d0e8d81d006b09319413b9fab5cff93c956/semantics/src/traversal/functions.rs#L168-L182

But I'm a bit confused, how do I lookup the function name of the function that hosts the `return`?

Reply:
Mmh..one way I can think of is to pass `name` into `func_stmt` and pass it on to each function that needs it.

https://github.com/ethereum/fe/blob/de661d0e8d81d006b09319413b9fab5cff93c956/semantics/src/traversal/functions.rs#L72-L74

Then it can be looked up from the context. Alternatively, instead of the `name` we could pass on the whole `FunctionAttributes` which would eliminate the need to do the lookup. Not sure if that's the right approach but I'm going to try it out to make progress on this.